models/volunteers.js

```diff
@@ -0,0 +1,81 @@
+var bookshelf = require('../libs/db').bookshelf;
+var constants = require('./constants.js');
+//var User = require('./user').User;
+var DrupalAccess = require('../libs/drupal_acces').DrupalAccess;
+var log = require('../libs/logger')(module);
+var VolunteerRole = bookshelf.Model.extend({
+    tableName: constants.VOL_DEPARTMENT_ROLES_TABLE_NAME,
+    volunteers: function() {
+        return this.belongsToMany(Volunteer, 'id', 'role_id');
+    }
+});
+
+var TypeInShift = bookshelf.Model.extend({
+    tableName: constants.VOL_TYPES_IN_SHIFT_TABLE_NAME
+});
+
+//Department
+var Department = bookshelf.Model.extend({
+    tableName: constants.VOL_DEPARTMENTS_TABLE_NAME,
+    volunteers: function() {
+        return this.hasMany(Volunteer, 'department_id');
+    },
+    shifts: function() {
+        return this.hasMany(Shift, 'id')
+    }
+});
+//Volunteer
+var Volunteer = bookshelf.Model.extend({
+    tableName: constants.VOLUNTEERS_TABLE_NAME,
+    department: function() {
+        return this.belongsTo(Department, 'department_id');
+    },
+    role: function() {
+        return this.belongsTo(VolunteerRole, 'role_id', 'id');
+    },
+    type_in_shift: function() {
+        return this.belongsTo(TypeInShift, 'type_in_shift_id');
+    },
+    user_info: function(user) {
+        return DrupalAccess.get_user_info(user_id);
+    }
+}, {
+    get_by_user: function(user_id_, dep_id, event_id) {
+        log.debug('Looking for volunteer data for user ' + user_id_);
+        return new Volunteer().fetch({ user_id: user_id_, department_id: dep_id, event_id: event_id }).then(function(vol) {
```

Comment:
in case of same name istead of `{keyname: var}` you can just `{keyname}`, example:
`{ user_id, department_id, event_id }` (change the function params accordingly).